PyGame_Project/MVC/Controller/CLI_controller.py

```diff
@@ -380,8 +420,68 @@ def start_shared_game():
         return start_shared_game()
       elif key == 'n':
         return True
-
   
   activeGameLoop()
 
+
+# for when the user considers giving up their current puzzle
+def give_up():
+  answer = user_input(0).lower()
+  match answer:
+    # you gave up!
+    case "y":
+      print_enter_name()
+      player_name = user_input(3)
+      all_letters = PuzzleStats().getPangram()
+      req_letter = PuzzleStats().getReqLetter()
```

Comment:
You should be able to do PuzzleStats().required_letter to retrieve all the letters.